common/types.go

```diff
@@ -234,16 +237,302 @@ func init() {
 }
 
 // Address represents the 20 byte address of an Ethereum account.
-type Address [AddressLength]byte
+type InternalAddress [AddressLength]byte
+type ExternalAddress [AddressLength]byte
+
+type Address struct {
+	inner AddressData
+}
+
+type AddressData interface {
+	Bytes() []byte
+	Hash() Hash
+	Hex() string
+	String() string
+	checksumHex() []byte
+	Format(s fmt.State, c rune)
+	SetBytes(b []byte)
+	MarshalText() ([]byte, error)
+	UnmarshalText(input []byte) error
+	UnmarshalJSON(input []byte) error
+	Scan(src interface{}) error
+	Value() (driver.Value, error)
+	Location() *Location
+	IsInChainScope() bool
+}
+
+func NewAddress() Address {
+	var a Address
+	var i InternalAddress
+	a.inner = &i
+	return a
+}
+
+// IsInChainScope checks if an address is a valid account in our node's sharded address space
+func (a Address) IsInChainScope() bool {
+	if a.inner == nil || a.inner.Hash() == ZeroAddr.Hash() {
+		return true
+	}
+	return NodeLocation.ContainsAddress(a)
+}
+
+func (a *Address) SetZero() *InternalAddress {
```

Comment:
is this a one time only startup call or used in many places?.

Reply:
It's not used anywhere actually. Removed

Reply:
Thanks